x-pack/plugins/secops/public/containers/uncommon_processes/index.tsx

```diff
@@ -71,60 +74,51 @@ const UncommonProcessesComponentQuery = pure<UncommonProcessesProps>(
           to: endDate,
         },
         pagination: {
-          limit,
+          limit: upperLimit,
           cursor,
           tiebreaker: null,
         },
         filterQuery,
       }}
     >
-      {({ data, loading, fetchMore, refetch }) =>
-        children({
+      {({ data, loading, refetch, updateQuery }) => {
+        const uncommonProcesses = getOr([], 'source.UncommonProcesses.edges', data);
+        const pageInfo = getOr(
+          { endCursor: { value: '' } },
+          'source.UncommonProcesses.pageInfo',
+          data
+        );
+
+        let endCursor = String(limit);
+        if (!isEmpty(pageInfo.endCursor.value)) {
+          endCursor = pageInfo.endCursor.value;
+        }
+
+        const hasNextPage = hasMoreData(parseInt(endCursor, 10), upperLimit, uncommonProcesses);
+        const slicedData = uncommonProcesses.slice(0, parseInt(endCursor, 10));
+        return children({
           id,
           loading,
           refetch,
           totalCount: getOr(0, 'source.UncommonProcesses.totalCount', data),
-          uncommonProcesses: getOr([], 'source.UncommonProcesses.edges', data),
-          pageInfo: getOr({}, 'source.UncommonProcesses.pageInfo', data),
+          uncommonProcesses: slicedData,
+          pageInfo: { hasNextPage, endCursor: { value: String(parseInt(endCursor, 10) + limit) } },
           loadMore: (newCursor: string) =>
-            fetchMore({
-              variables: {
-                pagination: {
-                  cursor: newCursor,
-                  limit,
-                },
-              },
-              updateQuery: (prev, { fetchMoreResult }) => {
-                if (!fetchMoreResult) {
-                  return prev;
-                }
-                return {
-                  ...fetchMoreResult,
-                  source: {
-                    ...fetchMoreResult.source,
-                    UncommonProcesses: {
-                      ...fetchMoreResult.source.UncommonProcesses,
-                      edges: [
-                        ...prev.source.UncommonProcesses.edges,
-                        ...fetchMoreResult.source.UncommonProcesses.edges,
-                      ],
-                    },
-                  },
-                };
-              },
-            }),
-        })
-      }
+            updateQuery(prev =>
+              set('source.UncommonProcesses.pageInfo.endCursor.value', newCursor, prev)
+            ),
+        });
+      }}
     </Query>
   )
 );
 
-const mapStateToProps = (state: State) => {
-  // TODO: This is hard coded without a reducer and state until
-  // we can determine if we can get a cursor object with the aggregate or not
-  // of uncommon_processes
-  const limit = 5;
-  return { limit };
-};
+export const hasMoreData = (
```

Comment:
Optional, you can have done this in a selector.
so you did not have to export it